src/VisualStudio/Core/Def/Implementation/ProjectSystem/MiscellaneousFilesWorkspace.cs

```diff
@@ -237,7 +237,15 @@ private void Registration_WorkspaceChanged(object sender, EventArgs e)
         {
             var workspaceRegistration = (WorkspaceRegistration)sender;
             uint docCookie;
-            Contract.ThrowIfFalse(_docCookieToWorkspaceRegistration.TryGetKey(workspaceRegistration, out docCookie));
+
+            // external workspace that listens to events from IVSRunningDocumentTable4 (just like MiscellaneousFilesWorkspace does) and registers the text buffer for opened document prior to Miscellaneous workspace getting notified of the docCookie - the prior contract assumes that MiscellaneousFilesWorkspace is always the first one to get notified
+            if (!_docCookieToWorkspaceRegistration.TryGetKey(workspaceRegistration, out docCookie))
```

Comment:
This is an edit to this code, and definitely won't regress it. Hopefully @mavasani can clarify why it was changed, since I don't understand the comments here.